vllm/engine/llm_engine.py

```diff
@@ -203,8 +208,31 @@ def _init_tokenizer(self, **tokenizer_init_kwargs):
             trust_remote_code=self.model_config.trust_remote_code,
             revision=self.model_config.tokenizer_revision)
         init_kwargs.update(tokenizer_init_kwargs)
-        self.tokenizer: TokenizerGroup = TokenizerGroup(
-            self.model_config.tokenizer, **init_kwargs)
+
+        if self.parallel_config.num_tokenizer_actors > 0:
+            if not RayTokenizerGroupPool:
+                raise ImportError(
+                    "RayTokenizerGroupPool is not available. Please install "
+                    "the ray package to use the tokenizer actors or "
+                    "set `num_tokenizer_actors` to 0.")
+            ray_actor_options = (self.parallel_config.tokenizer_actor_options
+                                 or {
+                                     "num_cpus": 0
+                                 })
+            ray_actor_options.setdefault(
+                "scheduling_strategy",
+                NodeAffinitySchedulingStrategy(
+                    node_id=ray.get_runtime_context().get_node_id(),
+                    soft=True))
+
+            init_kwargs[
+                "num_actors"] = self.parallel_config.num_tokenizer_actors
+            init_kwargs["ray_actor_options"] = ray_actor_options
+            self.tokenizer: BaseTokenizerGroup = RayTokenizerGroupPool(
+                self.model_config.tokenizer, **init_kwargs)
+        else:
+            self.tokenizer: TokenizerGroup = TokenizerGroup(
+                self.model_config.tokenizer, **init_kwargs)
```

Comment:
Can we move this complexity to a helper function / file? If we have a function like `create_tokenizer_group` we can unit test that correctly configures the ray actor / correctly returns the default tokenizer group / correctly throws an importerror if the ray tokenizer class is not defined.